src/sdkRuntimeModels.ts

```diff
@@ -217,14 +200,49 @@ export interface MParticleWebSDK {
         eventOptions?: SDKEventOptions
     ): void;
     logBaseEvent(event: BaseEvent, eventOptions?: SDKEventOptions): void;
+    logError(error: IErrorLogMessage, attrs?: SDKEventAttrs): void;
+    logLink(selector: string, eventName: string, eventType: valueof<typeof EventType>, eventInfo: SDKEventAttrs): void;
+    logForm(selector: string, eventName: string, eventType: valueof<typeof EventType>, eventInfo: SDKEventAttrs): void;
+    logPageView(eventName: string, attrs?: SDKEventAttrs, customFlags?: SDKEventCustomFlags, eventOptions?: SDKEventOptions): void;
+    setOptOut(isOptingOut: boolean): void;
     eCommerce: SDKECommerceAPI;
-    logLevel: string;
-    ProductActionType: SDKProductActionType;
+    isInitialized(): boolean;
+    ProductActionType: valueof<typeof ProductActionType>;
+    ready(f: Function): void;
+    reset(instance: IMParticleWebSDKInstance): void;
+    setAppName(name: string): void;
+    setAppVersion(version: string): void;
+    setOptOut(isOptingOut: boolean): void;
+    startTrackingLocation(callback?: Callback): void;
```

Comment:
Can you add a work item to audit callbacks and create interfaces for each of them?